### PR TITLE
FEATURE: DSS-2752 support for isolated domains

### PR DESCRIPTION
**Pull request by PetrVales**

### CODE REVIEW

File: src/main/java/com/gooddata/http/client/LoginSSTRetrievalStrategy.java [hunk 1]
```diff
@@ -60,11 +63,26 @@
...
+     * @param domain domain of user login
+     */
+    @Deprecated
+    public LoginSSTRetrievalStrategy(final HttpClient httpClient, final HttpHost httpHost, final String login, final String password, final String domain) {
         notNull(httpHost, "HTTP host cannot be null");
         notNull(login, "Login cannot be null");
         notNull(password, "Password cannot be null");
         this.login = login;
         this.password = password;
+        this.domain = null;
```
Comment (jakub-plichta) [inline src/main/java/com/gooddata/http/client/LoginSSTRetrievalStrategy.java]: `null` ?

File: src/main/java/com/gooddata/http/client/GoodDataHttpClient.java [hunk 2]
```diff
@@ -153,12 +169,24 @@ public GoodDataHttpClient(final SSTRetrievalStrategy sstStrategy) {
...
     public GoodDataHttpClient(final HttpClient httpClient, final HttpHost authHost, final SSTRetrievalStrategy sstStrategy) {
+        this(httpClient, authHost, sstStrategy, null);
+    }
+
+    /**
+     * Construct object.
+     * @param httpClient Http client
+     * @param authHost http host
+     * @param sstStrategy super-secure token (SST) obtaining strategy
+     * @param gdcHostHeader X-GDC_HOST header value
```
Comment (liry) [inline src/main/java/com/gooddata/http/client/GoodDataHttpClient.java]: Please unify naming and explanation in javadoc with one in gooddata-java (https://github.com/martiner/gooddata-java/pull/317)

Comment (PetrVales) [inline src/main/java/com/gooddata/http/client/GoodDataHttpClient.java]: I'll change it in gooddata-java.

File: src/test/java/com/gooddata/http/client/TestUtils.java [hunk 3]
```diff
@@ -67,9 +67,9 @@ static void logout(HttpClient client, HttpHost httpHost, String profile, int exp
         }
     }
 
-    static HttpClient createGoodDataClient(String login, String password, HttpHost host) {
+    static HttpClient createGoodDataClient(String login, String password, HttpHost host, String gdcHostHeader) {
         final HttpClient httpClient = HttpClientBuilder.create().build();
-        final SSTRetrievalStrategy sstStrategy = new LoginSSTRetrievalStrategy(login, password);
-        return new GoodDataHttpClient(httpClient, host, sstStrategy);
+        final SSTRetrievalStrategy sstStrategy = new LoginSSTRetrievalStrategy(login, password, gdcHostHeader);
+        return new GoodDataHttpClient(httpClient, host, sstStrategy, gdcHostHeader);
```
Comment (liry) [inline src/test/java/com/gooddata/http/client/TestUtils.java]: The usage is not very friendly (passing the value to strategy and then to client). Consider simplification to use it only once.

**Comment by liry** [comment]
What about to add general support for adding of any header instead of this concrete one?

**Comment by sadam21** [comment]
What is the purpose of this change? Does it make sense using `X-GDC_HOST` header somewhere other than PI?

**Comment by liry** [comment]
please change commit message - remove "FEATURE: DSS-2752" prefix

**Pull request closed without merging**